client/provider.go

```diff
@@ -111,9 +125,28 @@ func (cfg *providerConfig) validate() error {
 	} else if cfg.updateInterval.Seconds() < minUpdateInterval.Seconds() {
 		return errors.Errorf("update interval too small, minimum %v", minUpdateInterval)
 	}
+	if cfg.allowHTTP && strings.HasPrefix(cfg.url, "https://") {
+		return errors.Errorf("connecting to https endpoint: %s over gRPC/h2c, please unset AllowHTTP option", cfg.url)
+	} else if !cfg.allowHTTP && strings.HasPrefix(cfg.url, "http://") {
+		return errors.Errorf("connecting to http endpoint: %s over gRPC/TLS, please set AllowHTTP option", cfg.url)
+	}
```

Comment:
this validation doesn't work because our default sidecar URL also is prefixed with `https` - https://github.com/lekkodev/go-sdk/blob/main/client/api_provider.go#L37

I'm thinking we just remove it